scrunch/datasets.py

```diff
@@ -2034,275 +1454,67 @@ def export(self, path, format='csv', filter=None, variables=None,
-                dict(minBaseSize=dict(value=min_base_size)))
-        if palette:
-            payload['body'].setdefault('display_settings', {}).update(
-                dict(palette=palette))
-
-        # create the boxdata
-        self.resource.boxdata.create(payload)
-
-        # NOTE: the entity from the response is a bit different compared to
-        # others, i.e. no id, no delete method, different entity_url...
-        # For now, return the shoji_tuple from the index
-        for shoji_tuple in self.resource.boxdata.index.values():
-            if shoji_tuple.metadata.title == title:
-                return CrunchBox(shoji_tuple, self)
-
-    def delete_crunchbox(self, **kwargs):
-        """ deletes crunchboxes on matching kwargs """
-        match = False
-        for key in kwargs:
-            if match:
-                break
-            for crunchbox in self.crunchboxes:
-                attr = getattr(crunchbox, key, None)
-                if attr and attr == kwargs[key]:
-                    crunchbox.remove()
-                    match = True
-                    break
-
-
-class DatasetSubvariablesMixin(DatasetVariablesMixin):
-    """
-    Handles a variable subvariables iteration in a dict-like way
-    """
-
-    def __getitem__(self, item):
-        # Check if the attribute corresponds to a variable alias
-        subvariable = self.resource.subvariables.by('alias').get(item)
-        if subvariable is None:
-            # subvariable doesn't exists, must raise a ValueError
-            raise KeyError(
-                'Variable %s has no subvariable with an alias %s' % (self.name,
-                                                                     item))
-        # subvariable exists!, return the subvariable Instance
-        return Variable(subvariable, self)
-
-    def __iter__(self):
-        """
-        Iterable of ordered subvariables
-        """
-        sv_index = self.resource.subvariables.index
-        for sv in self.resource['body']['subvariables']:
-            yield sv_index[sv]
-
-    def __len__(self):
-        return len(self.resource.subvariables.index)
+    def get_exclusion(self):
+        # make sure there is an expression
+        expr = self.resource.exclusion.get('body').get('expression')
+        if not expr:
+            return None
+        return prettify(self.resource.exclusion.body.expression, self)
 
-    def itervalues(self):
-        for _, var_tuple in self.resource.subvariables.index.items():
-            yield Variable(var_tuple, self)
+    def add_filter(self, name, expr, public=False):
+        payload = dict(element='shoji:entity',
+                       body=dict(name=name,
+                                 expression=process_expr(parse_expr(expr), self.resource),
+                                 is_public=public))
+        new_filter = self.resource.filters.create(payload)
+        return self.filters[new_filter.body['name']]
```

Comment:
Subentities like Filter etc... are initialized with a ShojiTuple and not a ShojiEntity, this is so to avoid the roundtrip to Crunch and make it only if the user tries to access any attribute not present in the Tuple.